parsl/monitoring/db_local.py

```diff
@@ -0,0 +1,218 @@
+
+                    # check to make sure it is a task log and not just a workflow overview log
+                    if info.get('task_id', None) is not None:
+                        # if this is the first sight of the task in the workflow, add it to the workflow table
+                        if len(con.execute(self.meta.tables['task'].select(self.meta.tables['task'].c.task_id == info['task_id'])
+                                               .where(self.meta.tables['task'].c.task_run_id == run_id)).fetchall()) == 0:
+                            if 'psutil_process_pid' in info.keys():
+                                # this is the race condition that a resource log is before a status log so ignore this resource update
+                                return
+
+                            workflow = self.meta.tables['task']
+                            ins = workflow.insert().values(**{k: v for k, v in info.items() if k in workflow.c})
+                            con.execute(ins)
+
+                        if 'task_status' in info.keys():
+                            # if this is the first sight of a task, create a task_status_table to hold this task's updates
+                            if 'task_status' not in self.meta.tables.keys():
+                                task_status_table = create_task_status_table(self.meta)
+                                self.meta.create_all(con)
+                                con.execute(task_status_table.insert().values(**{k: v for k, v in info.items() if k in task_status_table.c}))
+                            # if this status table already exists, just insert the update
+                            else:
+                                task_status_table = self.meta.tables['task_status']
+                                con.execute(task_status_table.insert().values(**{k: v for k, v in info.items() if k in task_status_table.c}))
+                            return
+
+                        if 'psutil_process_pid' in info.keys():
+                            # if this is a task resource update then handle that, if the resource table DNE then create it
+                            if 'task_resources' not in self.meta.tables.keys():
+                                task_resource_table = create_task_resource_table(self.meta)
+                                self.meta.create_all(con)
+                                con.execute(task_resource_table.insert().values(**{k: v for k, v in info.items() if k in task_resource_table.c}))
+                            # if this resource table already exists, just insert the update
+                            else:
+                                task_resource_table = self.meta.tables['task_resources']
+                                con.execute(task_resource_table.insert().values(**{k: v for k, v in info.items() if k in task_resource_table.c}))
+                            return
+
+            except Exception as e:
+                print(str(e))
```

Comment:
log this at ERROR level rather than printing it, along with an explanation (eg your code comment at the start of the loop) so that someone reading the log doesn't have to read the source code. Log the number of tries with it as that info is easily available.